heron/common/src/java/com/twitter/heron/common/utils/logging/ErrorReportLoggingHandler.java

```diff
@@ -75,13 +85,20 @@ public void publish(LogRecord record) {
     // Convert Log
     Throwable throwable = record.getThrown();
     if (throwable != null) {
-      StringWriter sink = new StringWriter();
-      throwable.printStackTrace(new PrintWriter(sink, true));
-      String trace = sink.toString();
       synchronized (ExceptionRepositoryAsMetrics.INSTANCE) {
+        // We would not include the message if already exceeded the exceptions limit
+        if (ExceptionRepositoryAsMetrics.INSTANCE.getExceptionsCount() >= exceptionsLimit) {
```

Comment:
Both this counter and `droppedExceptionsCount ` are per message payload. 

Every time we collect the metrics, we would invoke IMetric.getAndReset(), send the message out and these two counter will reset to 0.